app/models/assessment_details_review.rb

```diff
@@ -0,0 +1,131 @@
+# frozen_string_literal: true
+
+class AssessmentDetailsReview
+  include ActiveModel::Model
+  extend ActiveModel::Callbacks
+
+  ASSESSMENT_DETAILS = %i[
+    summary_of_work
+    additional_evidence
+    site_description
+    consultation_summary
+  ].freeze
+
+  define_model_callbacks :save
+
+  attr_accessor :planning_application, :status
+
+  validate :recommendation_not_accepted
+
+  before_save :set_statuses
+
+  ASSESSMENT_DETAILS.each do |assessment_detail|
+    define_method(assessment_detail) do
+      memoize(
+        assessment_detail,
+        planning_application.send("existing_or_new_#{assessment_detail}")
+      )
+    end
+
+    define_method("#{assessment_detail}_comment") do
+      memoize(
+        "#{assessment_detail}_comment",
+        send(assessment_detail).existing_or_new_comment
+      )
+    end
+
+    delegate(
+      :review_status,
+      "review_status=",
+      :entry,
+      "entry=",
+      to: assessment_detail,
+      prefix: true
+    )
+
+    delegate(:text, "text=", to: "#{assessment_detail}_comment", prefix: true)
+
+    validates(
+      "#{assessment_detail}_review_status",
```

Comment:
OK I don't know what this does.
*I'm in danger meme*

Reply:
```
validates :summary_of_works_review_status, presence: true, if: :status_complete?`
```